src/shogun/base/SGObject.cpp

```diff
@@ -1135,6 +1135,12 @@ void CSGObject::push_back(CDynamicObjectArray* array, CSGObject* value)
 	array->push_back(value);
 }
 
+CSGObject* CSGObject::get(CDynamicObjectArray* array, index_t index)
```

Comment:
why does this need to be part of the SGObject interface?

Reply:
this is similar to the CSGObject.push_back that i dont get either why it's part of SGObject

Reply:
it can live anywhere. Just wanted to avoid including the DynamicObjectArray header file. Suggestions ?

Reply:
why SGObject and not DynamicObjectArray...
this could be a static method of DynamicObjectArray or just a namespace function....

Reply:
in latter case you dont need to include the header file...

Reply:
cool!